src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -246,6 +244,7 @@ macro_rules! bootstrap_tool {
         ;
     )+) => {
         #[derive(PartialEq, Eq, Clone)]
+        #[allow(dead_code)]
```

Comment:
Specifically it is `Tool::JsonDocCk` and `Tool::JsonDocLint`. For some reason, the compiler thought that they were used before this PR, probably because of some false positive (?). I haven't found anything in this PR that would remove their usage.

In general, I don't think that this should be a FIXME. The `Tool` enum is generated by a macro, and not all tools need to be built using `tool_cmd`, so it may be unused. The macro uses these two enum variants in the `tool_exe` function, but it is probably ignored by the compiler since the usage is inside a macro.